openmls/src/treesync/diff.rs

```diff
@@ -356,14 +356,24 @@ impl<'a> TreeSyncDiff<'a> {
         ciphersuite: &Ciphersuite,
         mut path_secret: PathSecret,
         sender_index: LeafIndex,
-    ) -> Result<CommitSecret, TreeSyncDiffError> {
-        let subtree_path = self.diff.subtree_path(self.own_leaf_index, sender_index)?;
+    ) -> Result<CommitSecret, TreeSyncSetPathError> {
+        // We assume both nodes are in the tree
```

Comment:
Why would we do that? I'd prefer it if we were principled here and turn it into a library error on the caller side if we have checked there.